### DISCUSSION TITLE
Can't report a "ui-interaction" event error

### DISCUSSION THREAD

**Discussion by TarikCarvalho08** [twilio-professional-services/flex-project-template#591] [Q&A] [answered]
Hello,

I'm having this user interaction error and I couldn't find a way to fix it. Normally, it happens when hanging up or ending a call. It usually happens when hanging up or ending a call.

I took a look at the features that call these events when a call is disconnected/completed, but had no success in fixing it.

When the button is pressed and the error occurs, it displays several errors in the console, one after the other, but Flex continues to work normally.

Here are images of the errors:

**Flex ui interaction error logs**

![image](https://github.com/twilio-professional-services/flex-project-template/assets/92161663/9a257b31-896a-4316-ae24-9c977199ef1d)

**Attached a document with the complete information of a single error log**
[Flex ui-interaction error.log](https://github.com/user-attachments/files/16100734/Flex.ui-interaction.error.log)

Thanks for your help!

**Comment by TarikCarvalho08** [comment]
After some testing, I discovered that this error only occurs when the **Supervisor Barge Coach** feature is activated. No matter how many other features are activated, as long as this one is, the error sometimes occurs when hanging up or completing a call.

**Comment by dremin** [comment, marked as answer]
Hi @TarikCarvalho08! Thanks for reporting, we will look into this. However, this error can be safely ignored as it is harmless.

**Reply by TarikCarvalho08** [reply]
Hi @dremin, Thanks for the information!

**Reply by dremin** [reply]
No problem! I've submitted some improvements in #605 that should help here; feel free to try out that branch if you'd like.

**Reply by dremin** [reply]
@TarikCarvalho08 The root cause of the error has been resolved in Flex UI 2.8.3 if you would like to try that version.

**Comment by stjohnson44** [comment]
Hi @dremin. This is an issue we have encountered as well - the issue appears to be that it is hitting the rate limit of flex telemetry, and when happening frequently can degrade performance. In my investigation it appears to be connected to the use of useFlexSelector, which I am working on memoizing to see if any improvements are seen. We have also seen some pages re-rendering when they should not be loaded (for instance, the queues page  seems to be re-rendering even when on the agent desktop and an agent changes their status). Please keep me updated on any developments as your team looks into this.

**Reply by dremin** [reply]
@stjohnson44 Your findings regarding `useFlexSelector` do match what I have found looking internally. I've raised this issue to Engineering to see what can be improved on our end so that plugins don't need to work around this. For now, I suggest reducing the amount of updates being made in the Redux store where possible, which should reduce re-renders and generally help performance. For the referenced supervisor-barge-coach feature, the updates in #605 should help a lot, as it was more or less spamming Redux updates at wrapup before the updates.

Regarding your findings about the queues view updates, since I don't believe it is related to the template, I would recommend filing this via support, which can escalate to Engineering with the appropriate details.

**Reply by dremin** [reply]
@stjohnson44 Please try Flex UI 2.8.3, the telemetry issues should now be resolved!
